app/models/concerns/subform_callbacks.rb

```diff
@@ -0,0 +1,36 @@
+module SubformCallbacks
+  extend ActiveSupport::Concern
+
+  def create_on_stockit
+    response = Stockit::ItemDetailSync.create(self)
+    if response && (errors = response["errors"]).present?
+      errors.each { |key, value| self.errors.add(key, value) }
+    elsif response && (stockit_id = response["#{self.class.name.underscore}_id"]).present?
+      update_column(:stockit_id, stockit_id)
+    end
+  end
+
+  def update_on_stockit
+    Stockit::ItemDetailSync.update(self)
```

Comment:
Update should also to assign back the stockit_id because in the case where you call Update and the record doesn't exist in Stockit then stockit will create the Item and return the stockit_id to you.

so I think update and create can be the same method

Reply:
sure, it's done